instrumentation/classloading/src/main/java/io/opentelemetry/auto/instrumentation/classloading/ClassloadingInstrumentation.java

```diff
@@ -0,0 +1,78 @@
+package io.opentelemetry.auto.instrumentation.classloading;
+
+import static io.opentelemetry.auto.tooling.ByteBuddyElementMatchers.safeHasSuperType;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.isProtected;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.isStatic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.auto.tooling.Constants;
+import io.opentelemetry.auto.tooling.Instrumenter;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(Instrumenter.class)
+public final class ClassloadingInstrumentation extends Instrumenter.Default {
+  public ClassloadingInstrumentation() {
+    super("classloading");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return safeHasSuperType(named("java.lang.ClassLoader"));
+  }
+
+  @Override
+  public String[] helperClassNames() {
+    return new String[] {Constants.class.getName()};
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        isMethod()
+            .and(named("loadClass"))
+            .and(
+                takesArguments(1)
+                    .and(takesArgument(0, named("java.lang.String")))
+                    .or(
+                        takesArguments(2)
+                            .and(takesArgument(0, named("java.lang.String")))
+                            .and(takesArgument(1, named("boolean")))))
+            .and(isPublic().or(isProtected()))
+            .and(not(isStatic())),
+        ClassloadingInstrumentation.class.getName() + "$LoadClassAdvice");
+  }
+
+  public static class LoadClassAdvice {
+    @Advice.OnMethodEnter(skipOn = Advice.OnNonDefaultValue.class)
```

Comment:
What is this for?

Reply:
this tells byte buddy to skip the `loadClass()` method body if we return something non-null (since we want to short-circuit and return from `loadClass()` right away if we find what we're looking for in the bootstrap class loader

Reply:
oh, I see... cool.